Scenes/Entities/Crawler/crawler.gd

```diff
@@ -0,0 +1,59 @@
+extends Node2D
+class_name EnemyNode
+
+# --- Exported Properties ---
+@export var actions: int = 2                ## Number of actions the enemy can take in a single turn.
+@export var speed: float = 150.0            ## Movement speed of the enemy, slightly slower than the player.
+
+# --- Public Properties ---
+var current_cell: Vector2i                  ## Current cell position of the enemy in the grid.
+var target_cell: Vector2                    ## Target cell position for movement.
+var path: Array = []                        ## Array to hold the calculated path for movement.
+var path_index: int = 0                     ## Current index in the path array.
+var move: bool = false                      ## Flag indicating if the enemy is currently performing an action.
+
+# --- Private Properties ---
+@onready var _actions: int = actions
+
+# --- Signals ---
+signal moved(position: Vector2i)            ## Signal emitted when the enemy’s position is updated.
+signal turn_ended                           ## Signal emitted when the enemy's turn ends.
+
+# --- Built-in Callbacks ---
+# Called when the node is added to the scene. Connects to the player’s position update signal.
+func _ready() -> void:
+	global.player.turn_ended.connect(_on_player_moved)
+# Updates the enemy's target cell and recalculates the path when the player moves.
+func _on_player_moved() -> void:
+	global.pathfinder.set_point_solid(global.map.local_to_map(position), false)
+	path = global.calculate_path(position, global.player.position).slice(0, actions+1)
+	if path.size() > 0:
+		global.pathfinder.set_point_solid(path[path.size()-1], true)
+		move = true
+
+# Called every frame. Updates path highlights and moves the enemy along the defined path.
+func _process(delta: float) -> void:
+	# Enemy movement along the path to the player
+	if path.size() > path_index:
+		global.player.action = true
+		var target_position = global.map.map_to_local(path[path_index])
+		global_position = global_position.move_toward(target_position, delta * speed)
+		if position.distance_to(target_position) < speed * delta:
+			current_cell = global.map.local_to_map(position)
+			path_index += 1
+			moved.emit(current_cell)
+	elif path.size() != 0:
+		global.player.action = false
+		position = global.map.map_to_local(path[path.size() - 1])
+		path.clear()
+		path_index = 0
+		move = false
+		if _actions == 0:
+			_actions = actions
+			turn_ended.emit()
+	
+# --- Custom Methods ---
+## Resets the enemy's actions for a new turn.
+func start_turn():
```

Comment:
Another thing that could be moved to the actor class. We can do this in another pr tho.